### PR TITLE
Fix move group namespace and demo.launch.py

### LINKED ISSUE #68: Quickstart_in_rivz problem
### Description
im learning moveit2 in ros2-foxy.when i run `ros2 launch moveit2_tutorials demo.launch.py rviz_tutorial:=true` by following [https://github.com/ros-planning/moveit2_tutorials/blob/main/doc/quickstart_in_rviz/quickstart_in_rviz_tutorial.rst](url), an error happend.
I would be very thankful for your help!!

### Your environment
* ROS Distro: |Foxy|
* OS Version: e.g. Ubuntu 20.04
* maybe source?
*
*

### Steps to reproduce
demo and files in [https://github.com/ros-planning/moveit2_tutorials/blob/main/doc/quickstart_in_rviz/quickstart_in_rviz_tutorial.rst](url)

### Expected behaviour
when i run `ros2 launch moveit2_tutorials demo.launch.py rviz_tutorial:=true` by following [https://github.com/ros-planning/moveit2_tutorials/blob/main/doc/quickstart_in_rviz/quickstart_in_rviz_tutorial.rst](url), an error happend.

### Backtrace or Console output

lee@lee-VirtualBox:~/ws_moveit2$ ros2 launch moveit2_tutorials demo.launch.py rviz_tutorial:=true
[INFO] [launch]: All log files can be found below /home/lee/.ros/log/2021-04-10-18-23-32-079928-lee-VirtualBox-17769
[INFO] [launch]: Default logging verbosity is set to INFO
Task exception was never retrieved
future: <Task finished name='Task-2' coro=<LaunchService._process_one_event() done, defined at /opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py:271> exception=InvalidLaunchFileError('py')>
Traceback (most recent call last):
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_description_sources/any_launch_file_utilities.py", line 53, in get_launch_description_from_any_launch_file
    return loader(launch_file_path)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_description_sources/python_launch_file_utilities.py", line 68, in get_launch_description_from_python_launch_file
    return getattr(launch_file_module, 'generate_launch_description')()
  File "/home/lee/ws_moveit2/install/moveit2_tutorials/share/moveit2_tutorials/launch/demo.launch.py", line 71, in generate_launch_description
    run_move_group_node = Node(package='moveit_ros_move_group',
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch_ros/actions/node.py", line 196, in __init__
    normalized_params = normalize_parameters(parameters)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch_ros/utilities/normalize_parameters.py", line 181, in normalize_parameters
    normalized_params.append(normalize_parameter_dict(param))
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch_ros/utilities/normalize_parameters.py", line 162, in normalize_parameter_dict
    raise TypeError('Unexpected type for parameter value {}'.format(repr(value)))
**TypeError: Unexpected type for parameter value None**

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py", line 273, in _process_one_event
    await self.__process_event(next_event)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py", line 293, in __process_event
    visit_all_entities_and_collect_futures(entity, self.__context))
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 45, in visit_all_entities_and_collect_futures
    futures_to_return += visit_all_entities_and_collect_futures(sub_entity, context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 45, in visit_all_entities_and_collect_futures
    futures_to_return += visit_all_entities_and_collect_futures(sub_entity, context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 38, in visit_all_entities_and_collect_futures
    sub_entities = entity.visit(context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/action.py", line 108, in visit
    return self.execute(context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/actions/include_launch_description.py", line 125, in execute
    launch_description = self.__launch_description_source.get_launch_description(context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_description_source.py", line 84, in get_launch_description
    self._get_launch_description(self.__expanded_location)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_description_sources/any_launch_description_source.py", line 53, in _get_launch_description
    return get_launch_description_from_any_launch_file(location)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_description_sources/any_launch_file_utilities.py", line 56, in get_launch_description_from_any_launch_file
    raise InvalidLaunchFileError(extension, likely_errors=exceptions)
**launch.invalid_launch_file_error.InvalidLaunchFileError: Caught exception when trying to load file of format [py]: Unexpected type for parameter value None**

### PR DESCRIPTION
**Pull request by vatanaksoytezer**
- Ports #62, #63 and #64 to the new ROS2 conventions.
- Fixes #68.
- Includes move_group namespace fixes (see https://github.com/ros-planning/moveit2/pull/420)
- Add pre-commit

### CODE REVIEW

File: doc/moveit_cpp/src/moveit_cpp_tutorial.cpp [hunk 1]
```diff
@@ -92,11 +120,11 @@ int main(int argc, char** argv)
     visual_tools.publishAxisLabeled(target_pose1.pose, "target_pose");
     visual_tools.publishText(text_pose, "Goal Pose", rvt::WHITE, rvt::XLARGE);
     // Visualize the trajectory in Rviz
-    visual_tools.publishTrajectoryLine(plan_solution1.trajectory, joint_model_group_ptr);
+    /* visual_tools.publishTrajectoryLine(plan_solution1.trajectory, joint_model_group_ptr); */
     visual_tools.trigger();
 
     /* Uncomment if you want to execute the plan */
-    /* planning_components->execute(); // Execute the plan */
+    // planning_components->execute(); // Execute the plan */
```
Comment (JafarAbdi) [inline doc/moveit_cpp/src/moveit_cpp_tutorial.cpp:127]:
```suggestion
    /* planning_components->execute(); // Execute the plan */
```

**Comment by vatanaksoytezer** [comment]
in move_group_interface tutorial when an object is attached the robot, the paths generated are not not collision aware and attached object collides with the object in the scene

**Comment by MarqRazz** [comment]
@vatanaksoytezer This looks great! Thanks for pushing forward. Does your move_group_interface tutorial plan around the collision object at the end? Mine goes straight through and says success.
collision_error

**Comment by vatanaksoytezer** [comment]
Hi @MarqRazz, thanks! Yeah I observed the same behaviour, @JafarAbdi mentioned he would have a look at this. Maybe you can open an issue and tag me and jafar?

**Comment by JafarAbdi** [comment]
@vatanaksoytezer Could you please rebase on the main branch .?

**Review by MarqRazz — APPROVED** [review]

**Pull request merged**